Libraries/LibWeb/Encoding/TextEncoderStream.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2025, Luke Wilde <luke@ladybird.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/Streams/GenericTransformStream.h>
+
+namespace Web::Encoding {
+
+class TextEncoderStream final
+    : public Bindings::PlatformObject
+    , public Streams::GenericTransformStreamMixin {
+    WEB_PLATFORM_OBJECT(TextEncoderStream, Bindings::PlatformObject);
+    GC_DECLARE_ALLOCATOR(TextEncoderStream);
+
+public:
+    static WebIDL::ExceptionOr<GC::Ref<TextEncoderStream>> construct_impl(JS::Realm&);
+    virtual ~TextEncoderStream() override;
+
+    static FlyString const& encoding();
```

Comment:
This is why `TextEncoderCommon` should be its own file - we would then have a `TextEncoderCommonMixin` interface that both `TextEncoder` and `TextEncoderStream` inherit from to implement this method.